octoprint_mrbeam/scripts/update_script.py

```diff
@@ -404,30 +427,48 @@ def loadPluginTarget(archive, folder):
         raise RuntimeError("Could not unzip plugin repo - error: {}".format(e))
 
     # copy new dependencies to working directory
-    try:
-        shutil.copy2(
-            os.path.join(
-                plugin_extracted_path_folder, MAIN_SRC_FOLDER_NAME, "dependencies.txt"
-            ),
-            os.path.join(folder, "dependencies.txt"),
-        )
-    except IOError:
-        raise RuntimeError("Could not copy dependencies to working directory")
+    _copy_file_to_working_dir(
+        folder=folder,
+        plugin_extracted_path_folder=plugin_extracted_path_folder,
+        origin_file="dependencies.txt",
+        destination_file="dependencies.txt",
+    )
 
     # copy new update script to working directory
+    _copy_file_to_working_dir(
+        folder=folder,
+        plugin_extracted_path_folder=plugin_extracted_path_folder,
+        origin_file="scripts/update_script.py",
+        destination_file="update_script.py",
+    )
+
+    # copy new update script config to working directory
+    _copy_file_to_working_dir(
+        folder=folder,
+        plugin_extracted_path_folder=plugin_extracted_path_folder,
+        origin_file="scripts/update_script_config.py",
+        destination_file="update_script_config.py",
+    )
+
+    return zip_file_path
+
+
+def _copy_file_to_working_dir(
```

Comment:
I like, that you split this in a new function, but please add a Google Doc string

Reply:
It's added :)

Reply:
It's added